agent/csi/libstorage/csi_service_libstorage_idemp.go

```diff
@@ -122,33 +124,69 @@ func (d *driver) IsNodePublished(
 	pubInfo *csi.PublishVolumeInfo,
 	targetPath string) (bool, error) {
 
-	idVal, ok := id.Values["id"]
-	if !ok {
-		return false, errMissingIDKeyPath
-	}
+	var (
+		devPath      string
+		volMountPath string
+		rootPath     = d.config.GetString(apitypes.ConfigIgVolOpsMountRootPath)
+	)
 
-	// Request only volumes attached to this instance.
-	opts := &apitypes.VolumeInspectOpts{
-		Attachments: apitypes.VolAttReqWithDevMapForInstance,
-		Opts:        apiutils.NewStore(),
-	}
+	if pubInfo != nil {
+		token, ok := pubInfo.Values["token"]
+		if !ok {
+			return false, errMissingTokenKey
+		}
 
-	vol, err := d.client.Storage().VolumeInspect(d.ctx, idVal, opts)
-	if err != nil {
-		return false, err
-	}
+		// Get device from local devices
+		opts := &apitypes.LocalDevicesOpts{
+			Opts:     apiutils.NewStore(),
+			ScanType: apitypes.DeviceScanQuick,
+		}
+		devs, err := d.client.Executor().LocalDevices(d.ctx, opts)
+		if err != nil {
+			return false, fmt.Errorf("unable to get local devices")
```

Comment:
Hi @codenrhoden,

Please consider creating a package field like so:

```
var errUnableToGetLocDevs = errors.New("unable to get local devices")
```

This way a new error isn't allocated each time there is a failure.